gapic/src/main/com/google/gapid/perfetto/views/State.java

```diff
@@ -122,6 +129,28 @@ public long deltaPxToDuration(double px) {
     return Math.round(px * nanosPerPx);
   }
 
+  protected SortedMap<Double, Long> searchForFlag(double x) {
+    return flags.subMap(x - 12, x + 12);
```

Comment:
Currently, there's a UX problem that when a flag is created and some zoomings happens afterwards, it may be difficult to remove a flag. I think it is related to this fixed number 12, I would suggest using something like 3 * deltaPxToDuration(1) instead, to denote a dynamic 3 pixel flexible range.

Reply:
Done